vars/runATH.groovy

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env groovy
+
+/**
+ * Simple wrapper for running the ATH
+ */
+
+def call(Map params = [:]) {
+
+    def String athUrl = params.get('athUrl', 'https://github.com/jenkinsci/acceptance-test-harness.git')
+    def String athRevision = params.get('athRevision', 'master')
+    def String metadataFile = params.get('metadataFile', 'essentials.yml')
+    def String jenkins = params.get('jenkins', 'latest')
+
+    def isLocalATH = athUrl.startsWith("file://")
+    def isVersionNumber = (jenkins =~ /^(\d+\.)?(\d+\.)?(\*|\d+)$/).matches()
+    // Workaround for https://issues.jenkins-ci.org/browse/JENKINS-27092
+    def shouldStop = false
+    // End of workaround
+
+    def mirror = "http://mirrors.jenkins.io/"
+    def defaultCategory = "org.jenkinsci.test.acceptance.junit.SmokeTest"
+    def jenkinsURl = jenkins
+    def metadata
+    def athContainerImage
+
+
+    def athSourcesFolder = "athSources"
+
+    stage("Getting ATH sources and Jenkins war") {
+        // Start validation
+        if (!fileExists(metadataFile)) {
+            echo "The metadata file does not exist. Current value is ${metadataFile}"
+            shouldStop = true
+            return
+        } else {
+            metadata = readYaml(file: metadataFile).ath
+            if (metadata == null) {
+                error "The provided metadata file seems invalid as it does not contain the ath section"
+            }
+            if (metadata.browsers == null) {
+                echo "The provided metadata file does not include the browsers property, using firefox as default"
+            }
+        }
+        if (!isLocalATH) {
+            def canConnectToGitRepo = sh(script: "git ls-remote --exit-code -h ${athUrl}", returnStatus: true) == 0
+            if (!canConnectToGitRepo) {
+                error "Is not possible to connect to the given athUrl, please review it. Current value for athUrl is ${athUrl}"
+            }
```

Comment:
```groovy
echo 'Checking connectivity to ATH sources…'
sh "git ls-remote --exit-code -h ${athUrl}"
```

would suffice I think.

Reply:
Yeah, I believe my version is more user-friendly but probably you are right and I am being too much user-friendly here without a compelling reason